build_tools/check_backticks.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Test script to check for invalid use of single-backticks."""
+
+__author__ = ["geetu040"]
+
+import ast
+import os
+import re
+
+
+def find_py_files(folder_path):
+    """Find all Python files in a given folder path."""
+    py_files = []
+    for root, _, files in os.walk(folder_path):
+        for file in files:
+            if file.endswith(".py"):
+                py_files.append(os.path.join(root, file))
+    return py_files
+
+
+def extract_docstrings(filename):
+    """Extract docstrings from a Python file."""
+    with open(filename) as file:
+        tree = ast.parse(file.read(), filename=filename)
+
+    docstrings = {}
+    for node in ast.walk(tree):
+        if isinstance(node, (ast.FunctionDef, ast.ClassDef, ast.Module)):
+            if (
+                node.body
+                and isinstance(node.body[0], ast.Expr)
+                and isinstance(node.body[0].value, ast.Str)
+            ):
+                lineno = 0 if isinstance(node, ast.Module) else node.lineno
+                docstring = node.body[0].value.s
+                docstrings[lineno] = docstring
+    return docstrings
+
+
+def find_invalid_backtick_text(docstring):
+    """Find invalid backtick text in a docstring."""
+    # remove all the double-backticks
+    doc = docstring.replace("``", "")
+
+    all_backtick_text = re.findall(r"`.*?`", doc)
+    valid_backtick_text = re.findall(r":.*?:(`.*?`)", doc)
```

Comment:
```
:math:`d(x, y):= (x-y)^2`
```
expression like these are valid cases because they have special representation in documentation generated by `rst`

Reply:
This looks extremely specific.

Your code however matches for any `:`, not just ones after `:math:`. Will it make sense to check for that (and any other specific cases you want to allow), perhaps with positive look behind?

Reply:
Its not specific, there are multiple cases like
```
:attr:`means_`
:data:`sys.stderr`
:term:`Glossary <random_state>`
:ref:`User Guide <cross_validation>`
:math:`i_k \in \{1, \dots, m\}, j_k \in \{1, \dots, n\}`
```